rl4j/rl4j-core/src/main/java/org/deeplearning4j/rl4j/util/DataManagerSyncTrainingListener.java

```diff
@@ -0,0 +1,105 @@
+package org.deeplearning4j.rl4j.util;
+
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.rl4j.learning.sync.listener.SyncTrainingEpochEndEvent;
+import org.deeplearning4j.rl4j.learning.sync.listener.SyncTrainingEvent;
+import org.deeplearning4j.rl4j.learning.sync.listener.SyncTrainingListener;
+
+@Slf4j
+public class DataManagerSyncTrainingListener implements SyncTrainingListener {
+    private final IDataManager dataManager;
+    private final int saveFrequency;
+    private final int monitorFrequency;
+
+    private int lastSave;
+    private int lastMonitor;
+
+    private DataManagerSyncTrainingListener(Builder builder) {
+        this.dataManager = builder.dataManager;
+
+        this.saveFrequency = builder.saveFrequency;
+        this.lastSave = -builder.saveFrequency;
+
+        this.monitorFrequency = builder.monitorFrequency;
+        this.lastMonitor = -builder.monitorFrequency;
+    }
+
+    @Override
+    public void onTrainingStart(SyncTrainingEvent event) {
+        try {
+            dataManager.writeInfo(event.getLearning());
+        } catch (Exception e) {
+            log.error("Training failed.", e);
+            e.printStackTrace();
```

Comment:
Don't also print stack trace, log.error will already print it.
Also other instances below.